Purchases/Public/RCPurchases.m

```diff
@@ -299,7 +299,7 @@ - (instancetype)initWithAppUserID:(nullable NSString *)appUserID
 
         [self.systemInfo isApplicationBackgroundedWithCompletion:^(BOOL isBackgrounded) {
             if (!isBackgrounded) {
-                [self.operationDispatcher dispatchOnWorkerThread:^{
+                [self.operationDispatcher dispatchOnWorkerThreadWithRandomDelay:NO block:^{
```

Comment:
yeah, I'll separate into two methods. I'd started by doing that, then I moved to having the randomDelay be optional, then the optional randomDelay looked great in swift but ported very poorly to obj-c, so I killed the optionality. I'll go back to the start.

Reply:
ok, I was trying this and remembered why I ended up going with the param: it's so that callers don't have to choose between methods, since almost every time they pass in the value of `isBackgrounded`. 

So if there are two separate methods, you'd have to do `if isBackgrounded dispatchWithRandomDelay else dispatch`. 
note that we almost every time pass `isBackgrounded`. even in this line, we're passing `NO` because we know that `isBackgrounded` is false, because of the `if` in the line right above. But we could just as easily pass `isBackgrounded` instead.